src/GraphQL.Conventions/Adapters/Engine/GraphQLEngine.cs

```diff
@@ -213,15 +214,23 @@ internal IDependencyInjector DependencyInjector
             string operationName,
             Inputs inputs,
             IUserContext userContext,
+            ComplexityConfiguration complexityConfiguration,
             bool enableValidation = true,
             bool enableProfiling = false,
             IEnumerable<IValidationRule> rules = null,
             CancellationToken cancellationToken = default(CancellationToken))
+            
         {
             if (!enableValidation)
             {
                 rules = new[] { new NoopValidationRule() };
             }
+
+            complexityConfiguration = (query.ToLower().Contains("introspectionquery") &&
```

Comment:
@bourgoisa thanks for the update - could you please make the changes that @huysentruitw proposed as well, and I'll get this merged?